index.bs

```diff
@@ -568,13 +578,23 @@ When this method is invoked, the user agent MUST execute the following algorithm
 
 1. Let |issuedRequests| and |currentlyAvailableAuthenticators| be new [=ordered sets=].
 
-1. For each |authenticator| currently available on this platform, if
-    <code>|options|.{{MakeCredentialOptions/attachment}}</code> is [=present|not present=] or its value matches
-    |authenticator|'s [=attachment modality=], [=set/append=] |authenticator| to |currentlyAvailableAuthenticators|. 
-
 1. If |currentlyAvailableAuthenticators| [=list/is empty=], return a {{DOMException}} whose name is 
     "{{NotFoundError}}", and terminate this algorithm.
 
+1. For each [=authenticator=] currently available on this platform, process each option in 
+    |options|.{{MakeCredentialOptions/authenticatorSelection}} by checking if the [=authenticator=] satisfies the 
+    requirement of the option. 
+        1. Check if |attachment| is [=present|not present=] or its value matches
+            |authenticator|'s attachment modality, [=continue=].
+        1. If |requireResidentKey| is set to |true|, check if the |authenticator| is capable of storing 
+            [=Client-Side-Resident Credential Private Key=]. If so, [=continue=]. If not, return a {{DOMException}} whose name
+            is "{{QuotaExceededError}}".
+        1. If the [=authenticator=] is capable of all criterias as listed above, [=set/append=] [=authenticator=] to 
```

Comment:
If we got here in the substeps, "If the [=authenticator=] is capable of all criterias as listed above," is known to be true, so you don't need to say it.